src/emitter.rs

```diff
@@ -260,6 +252,26 @@ impl<'a> YamlEmitter<'a> {
         }
     }
 
+    fn emit_literal_block(&mut self, v: &String) -> EmitResult {
+        let ends_with_newline = v.chars().last() == Some('\n');
+        if ends_with_newline {
+            write!(self.writer, "|")?;
+        } else {
+            write!(self.writer, "|-")?;
+        }
```

Comment:
I do like how this is simple, but I wonder if it'd be better to mash these two branches down into a single expression by pushing the conditional down into the parameter, e.g.

```rust
        write!(self.writer, if ends_with_newline { "|" } else { "|-" })?;
```

Hmm.. yeah.. now that I read that, I think I like it better the way you have it written as-is, but if someone thinks otherwise I wouldn't mind it either way.